scipy/fft/_helper.py

```diff
@@ -98,4 +101,192 @@ def _init_nd_shape_and_axes(x, shape, axes):
         Axes along which the calculation is computed. It is a 1-D integer array.
 
     """
-    return _helper._init_nd_shape_and_axes(x, shape, axes)
+    xp = array_namespace(x)
+    x = np.asarray(x)
+    shape, axes = _helper._init_nd_shape_and_axes(x, shape, axes)
+    return xp.asarray(shape), xp.asarray(axes)
```

Comment:
This returns a list, it doesn't belong in an array.

Reply:
Okay, how would you recommend changing this? I was following the standard pattern for attempting to convert to `np` and back to `xp` when hitting compiled code, but sounds like we want something a bit different here.

Do the docs need to be changed? Under `Returns` we have `shape : array` and `axes : array`.

Reply:
Why would this private helper function need changes at all? It's used like so:
```
shape, axes = _init_nd_shape_and_axes(x, ...)
```
so it should be fine to keep the return types as they were? `shape` and `axes` can be tuples at least; lists as they are now isn't guaranteed to work with most functions that have `shape=`. So if something is choking on a list, maybe try to return tuples instead.

The docs don't need to change as far as I can tell.

Reply:
@peterbell10 @rgommers is this the desired change? https://github.com/scipy/scipy/pull/19005/commits/aed9f821389f2935e946948f95be13abd2981b9a

Reply:
Not quite. I can see where the confusion was coming from though, because the docs of the function were incorrect to begin with - those said the return values were arrays. Instead, `shape` was a list and `axes` was a `range` object. Both aren't ideal either - `shape` should be a tuple and `axes` a list.

In figuring that out, I have most of a fix ready, so let me see if I can push that here.

Reply:
done in commit 14cd80632f. and I also pushed a commit to include `fft` in the array API CI job.